From 30e74663b0cbb25e06eae3f61f0eaf753e98605d Mon Sep 17 00:00:00 2001
From: Dennis Hall <dennis.hall@windriver.com>
Date: Fri, 13 Dec 2019 00:28:51 -0800
Subject: [PATCH] Limit kbdrate to x86, mips and sparc.

    CQID: WIND00371274

    The application will exit with an error if it runs on any other architecture.

    The code relies on hardware specific memory locations to access
    and modify the keyboard repeat rate.  It also requires read/write
    access to /dev/port which doesn't exist on every architecture's
    root fs.  The CQ was raised for Qemu PowerPC but it also fails on
    ARM.  The keyboard emulation in qemuppc is for an ADB (Apple Desktop Bus)
    device and not compatible with an Intel driver.  There's also no
    indication in the documentation that the code should work on
    anything other than Intel architecture but it also works on MIPS32.
    MIPS64 is not supported at this time so it was not tested.

Signed-off-by: Dennis Hall <dennis.hall@windriver.com>

Upstream-Status: Inappropriate [wrlinux specific]
Rebase to 2.2.0
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Include header file libgen.h to fix error:

| ../../kbd-2.6.4/src/kbdrate.c:363:94: error: implicit declaration of function 'basename'; did you mean 'rename'? [-Wimplicit-function-declaration]
|   363 |                 "ERROR:  %s should only be used on MIPS, x86 and Sparc archicture boards\n", basename(argv[0]));
|       |                                                                                              ^~~~~~~~
|       |                                                                                              rename

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 src/kbdrate.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/kbdrate.c b/src/kbdrate.c
index 65baf20..bae8108 100644
--- a/src/kbdrate.c
+++ b/src/kbdrate.c
@@ -80,6 +80,7 @@ beats rebuilding the kernel!
 #include <sys/file.h>
 #include <sys/ioctl.h>
 #include <linux/kd.h>
+#include <libgen.h>
 
 #ifdef __sparc__
 #include <asm/param.h>
@@ -333,6 +334,16 @@ int main(int argc, char **argv)
 	int silent = 0;
 	int c;
 
+#if !(  defined(__i386__)   \
+     || defined(__x86_64__) \
+     || defined(__mips__)   \
+     || defined(__sparc___)) 
+
+	fprintf(stderr,
+		"ERROR:  %s should only be used on MIPS, x86 and Sparc archicture boards\n", basename(argv[0]));
+        exit(1);
+#endif
+
 	setuplocale();
 
 	const char *short_opts = "r:d:pshV";
-- 
2.47.0

